Packs/GigamonThreatINSIGHT/Integrations/GigamonThreatINSIGHT/GigamonThreatINSIGHT.py

```diff
@@ -0,0 +1,1142 @@
+
+    prefix = 'Insight.Entity.PDNS'
+    key = 'passivedns'
+
+    if not result:
+        raise Exception(f'We receive an invalid response from the server({result})')
+
+    if 'result_count' in result and result.get('result_count') == 0:
+        return "We could not find any result for Get Entity PDNS."
+
+    if key not in result:
+        raise Exception(f'We receive an invalid response from the server (The response does not contains the key: {key})')
+
+    if not result.get(key):
+        return "We could not find any result for Get Entity PDNS."
+
+    return CommandResults(
+        outputs_prefix=prefix,
+        outputs_key_field=key,
+        outputs=result.get(key)
+    )
+
+
+def commandGetEntityDhcp(entityClient: EntityClient, args: Dict[str, Any]):
+    """ Get DHCP information about an IP address.
+    """
+    demisto.debug('commandGetEntityDhcp has been called.')
+
+    entity = args.pop('entity')
+    result: Dict[str, Any] = entityClient.getEntityDhcp(entity, encodeArgsToURL(args, ['account_uuid']))
+
+    prefix = 'Insight.Entity.DHCP'
+    key = 'dhcp'
+
+    if not result:
+        raise Exception(f'We receive an invalid response from the server ({result})')
+
+    if 'result_count' in result and result.get('result_count') == 0:
+        return "We could not find any result for Get Entity DHCP."
+
+    if key not in result:
+        raise Exception(f'We receive an invalid response from the server (The response does not contains the key: {key})')
+
+    if not result.get(key):
+        return "We could not find any result for Get Entity DHCP."
+
+    return CommandResults(
+        outputs_prefix=prefix,
+        outputs_key_field=key,
+        outputs=result.get(key)
+    )
+
+
+def commandGetEntityFile(entityClient: EntityClient, hash: str):
+    """ Get entity information about a file
+    """
+    demisto.debug('commandGetEntityFile has been called.')
+
+    result: Dict[str, Any] = entityClient.getEntityFile(hash)
+
+    prefix = 'Insight.Entity.File'
+    key = 'file'
+
+    if not result:
+        raise Exception(f'We receive an invalid response from the server ({result})')
+
+    if key not in result:
+        raise Exception(f'We receive an invalid response from the server (The response does not contains the key: {key})')
+
+    if not result.get(key):
+        return "We could not find any result for Get Entity File."
+
+    return CommandResults(
+        outputs_prefix=prefix,
+        outputs_key_field=key,
+        outputs=result.get(key)
+    )
+
+
+# Detections API commands
+
+
+def commandFetchIncidents(detectionClient: DetectionClient, account_uuid, params, last_run) -> Tuple[Dict[str, int], List[dict]]:
+    demisto.debug('commandFetchIncidents has been called.')
+
+    demisto.debug(f'last_run retrieved: {last_run}')
+
+    firstFetch = getFirstFetch(params.get('first_fetch'))
+    first_fetch_time = datetime.now() - timedelta(days=firstFetch['days'], hours=firstFetch['hours'])
```

Comment:
There's a much easier way to retrieve the first fetch, you can refer to ***OKTA_IAM*** integration for a good example of how to easily parse the first_fetch param.
Also - first fetch is not necessarily only hours/days, it might be months, etc..